api/api.go

```diff
@@ -393,19 +357,15 @@ func loadYaml(path string) (string, error) {
 }
 
 // WhoamiQuery returns the result of querying the `/me` endpoint of the API
-func WhoamiQuery(ctx context.Context, logger *logger.Logger) (*WhoamiResponse, error) {
+func WhoamiQuery(ctx context.Context, cfg *settings.Config) (*WhoamiResponse, error) {
```

Comment:
I see a few (maybe all) of these `api` functions are accepting a `*settings.Config`, but they only use the client. Accepting a large structure or interface than is necessary for the function generally makes it harder to refactor later.

Could these functions accept just a `Client` ?